integrations/cdi/jta-cdi/src/main/java/io/helidon/integrations/jta/narayana/ProcessId.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) 2025 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.integrations.jta.narayana;
+
+import com.arjuna.ats.arjuna.utils.Process;
+
+/**
+ * Replacement for Arjuna's default {@code com.arjuna.ats.internal.arjuna.utils.SocketProcessId}
+ * which is not needed since JDK 9 introduced {@code ProcessHandle}.
+ * As no extra socket needs to be opened in this implementation, CRaC snapshotting is not affected.
+ */
+public class ProcessId implements Process {
```

Comment:
Maybe `final`?

Reply:
Why?

Reply:
Nothing big, just usually you want to err on the side of not designing for extensibility

Reply:
No harm in being extra safe, it's just the package visibility is limited to the Narayana only